3DSView/src/main/java/eu/livotov/labs/android/d3s/D3SView.java

```diff
@@ -248,11 +271,13 @@ private void completeAuthorization(String html)
             }
         }
 
-        if (authorizationListener != null)
+        // If we didn't find both MD and PaRes, don't continue
+        if (TextUtils.isEmpty(md) || TextUtils.isEmpty(pares)) {
```

Comment:
Your comment says "both MD and Pares", but your check is an OR. Is the OR correct?

Reply:
Yeah we need both so if 1 is empty, return.

Reply:
Even though I'm not the author, I'll jump in my my 2c.

Yes, I believe the OR is correct. Both pieces of information are needed to continue with the payment.

Caveat: I'm not 100% sure what I'm about to say is correct - I believe the MD is some sort of reference for the verification process that's passed around, so it's _conceivable_ the MD could be present without the PARes (the important bit that signifies the end of the process ... the Payer Authentication Response).

TL;DR: Unless I see evidence suggesting otherwise, I'd vote for keeping the code as is.

Reply:
Thanks, yes the OR makes sense thinking about it, just wanted to confirm

Reply:
Follow up - I've done a quick bit of research and apparently the MD parameter is optional according to some specifications, eg:

https://developer.paysafe.com/en/rest-apis/3ds/typical-api-calls/submit-a-request-to-the-3d-secure-acs-server-and-retrieve-a-pares/

Though other specifications I have found say it's required. So, conflicting information.